### ISSUE TITLE
RangeError on parseHash

### ISSUE THREAD

**Issue by steven89** [clayzermk1/node-marshal#27]
calling this on a MySQL blob:
```javascript
new Marshal(data.blob.toString(), 'binary')
```
I get this error for some of the records:
```
RangeError [ERR_OUT_OF_RANGE]: The value of "offset" is out of range. It must be >= 0 and <= 276. Received 277
    at new NodeError (node:internal/errors:371:5)
    at boundsError (node:internal/buffer:86:9)
    at Buffer.readUInt8 (node:internal/buffer:252:5)
    at Marshal._parse (/.../node_modules/marshal/lib/marshal.js:19:32)
    at Marshal._parseHash (/.../node_modules/marshal/lib/marshal.js:247:20)
    at Marshal._parse (/.../node_modules/marshal/lib/marshal.js:45:21)
    at Marshal.load (/.../node_modules/marshal/lib/marshal.js:345:26)
    at new Marshal (/.../node_modules/marshal/lib/marshal.js:13:33)
    ... {
  code: 'ERR_OUT_OF_RANGE'
}
```

The original ruby object is a Hashmap, is this package limited to a certain size of hash?
Unfortunately I cannot provide reproducible data at the time as it contains real user data. 
I will try to generate random Hashmaps later to see if I can reproduce this and give you data.

EDIT:
A bit more info
node version: `v16.13.2`
the MySQL `OCTET_LENGTH` gives `277` for the record that generated this particular error.

**Comment by clayzermk1** [comment]
The off-by-one length smells like a bug. I'd be keen on seeing that repro
data. Let me know!

**Comment by steven89** [comment]
The error doesn't appear when directly passing the `Buffer`
```javascript
new Marshal(data.blob)
```
I didn't notice that the original buffer object could be used.
As a suggestion maybe the `Buffer` input could be in the README example, had to go look in the code to figure it out.

I did find a similar issue in the `mysql2` repo https://github.com/sidorares/node-mysql2/issues/989 was solved by this commit https://github.com/sidorares/node-mysql2/commit/6dd8a4d8ba16bf5ee724c55bcc8c05f57026dd99 something to do with "session track block"
Maybe the issue encountered here is similar, the string representation of a buffer from mysql has to be parsed differently

As a side note, looking at the code it seems weird that the conversion of buffer to string and back to buffer would introduce such a bug.
